### PR TITLE
JS: add closure library in globalObjectRef

### PR DESCRIPTION
**Pull request by asger-semmle**
Recognizes [goog.global](https://google.github.io/closure-library/api/goog.html#global) as an alias for the global object.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/dataflow/Nodes.qll [hunk 1]
```diff
@@ -284,6 +284,9 @@ DataFlow::SourceNode globalObjectRef() {
   or
   // `require("global")`
   result = moduleImport("global")
+  or
+  // Closure library - based on AST to avoid recursion with Closure library model
+  result.asExpr().(DotExpr).getQualifiedName() = "goog.global"
```
Comment (xiemaisi) [inline javascript/ql/src/semmle/javascript/dataflow/Nodes.qll]: Do we perhaps want to use `globalVarRef` to ensure `goog` is a global and not a local of the same name?

**Comment by asger-semmle** [comment]
[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/augustus.ti.semmle.com_1551195883570). Currently re-running on box2d. The "failures" seem to be snapshots that aren't part of closure.slugs, so not related.

**Comment by asger-semmle** [comment]
Looks fine after [re-running](https://git.semmle.com/asger/dist-compare-reports/tree/augustus.ti.semmle.com_1551199708591).

**Comment by xiemaisi** [comment]
Excellent, thanks. Merging.

@hvitved, can we do something about the spurious "failed builds" in the report? (I think you added that feature.)

**Pull request merged**